### ISSUE TITLE
Provide a way to override the output location of the assets.json file

### ISSUE THREAD

**Issue by mellinoe** [dotnet/sdk#1057]
It seems that right now, the assets.json file is always output into [`$(BaseIntermediateOutputPath)`](https://github.com/dotnet/sdk/blob/b76e22d01bd8ce8ce0b0997a6800e804986bf29c/src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.PackageDependencyResolution.targets#L28). NuGet respects a different property, [`$(RestoreOutputPath)`](https://github.com/NuGet/NuGet.Client/blob/68245312b050c5b992120448353eba3e20cdc414/src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets#L422-L429 ).

I would just like a single, well-supported way to override the location.

@eerhardt

**Comment by dsplaisted** [comment]
RestoreOutputPath is set based on BaseIntermediateOutputPath.  Is there a reason not to just set BaseIntermediateOutputPath?

Note that you need to do so before the common .props are imported, which you can do by making the SDK imports explicit or by using a Directory.Build.props file.

**Comment by eerhardt** [comment]
> Is there a reason not to just set BaseIntermediateOutputPath?

In dotnet/corefx (and any other repos that are build-tools based), $(BaseIntermediateOutputPath) is not unique for every project.  Instead it is set to something like `$(RepoRoot)\obj`.  And then each project appends its name in $(IntermediateOutputPath).

See 
https://github.com/dotnet/corefx/blob/3ed853aaa7eaabc418f60bd797e19b876fce9b5a/dir.props#L21-L28
and
https://github.com/dotnet/corefx/blob/3ed853aaa7eaabc418f60bd797e19b876fce9b5a/dir.props#L231-L232

Thus putting `project.assets.json` in the `root\obj` folder isn't going to work, since it is not unique across all projects.

The fix here should be as easy as changing https://github.com/dotnet/sdk/blob/b76e22d01bd8ce8ce0b0997a6800e804986bf29c/src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.PackageDependencyResolution.targets#L28 to use `$(RestoreOutputPath)` instead.

**Comment by eerhardt** [comment]
@dsplaisted - don't you need to update

https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.PackageDependencyResolution.targets#L28

?  Or at least remove it?

**Comment by dsplaisted** [comment]
@eerhardt #2010 does that

**Comment by crusader-mike** [comment]
> The assets file will go in the `MSBuildProjectExtensionsPath`, which should probably be unique per project.

No, if two C# projects files are in the same directory. I just spent a lot of time trying to figure out why my build went completely nuts after I migrated to PackageReference.
